File: 0107/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<List<Integer>> levelOrderBottom(TreeNode root) {
        LinkedList<List<Integer>> r = new LinkedList<List<Integer>>();
        LinkedList<TreeNode> queue = new LinkedList<TreeNode>();
        int cnt = 1;
        if(root != null) {
            queue.add(root);
        }
        
        while(!queue.isEmpty()) {
            LinkedList<Integer> ele = new LinkedList<Integer>();
            while(cnt != 0) {
                TreeNode n = queue.poll();
                ele.add(n.val);
                if(n.left != null) {
                    queue.add(n.left);
                }
                if(n.right != null) {
                    queue.add(n.right);
                }
                cnt--;
            }
            cnt = queue.size();
            r.addFirst(ele);
        }
        
        return r;
    }
}

File: 0107/Solution.cpp
#include <iostream>
#include <string>
#include <vector>
#include <unordered_map>
#include <unordered_set>
#include <queue>
#include <algorithm>

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector<vector<int>> ans;
        queue<TreeNode *> queue;
        
        if (!root)
            return ans;

        queue.push(root);
        while (queue.size()) {
            vector<int> ans_r;
            int q_sz = queue.size();
            for (int i = 0; i < q_sz; i ++) {
                TreeNode *n = queue.front();
                queue.pop();
                ans_r.push_back(n->val);
                if (n->left)
                    queue.push(n->left);
                if (n->right)
                    queue.push(n->right);
            }
            ans.push_back(ans_r);
        }

        reverse(ans.begin(), ans.end());
        return ans;
    }
};

int main(int argc, char *argv[]) {
    Solution solution;
    deque<TreeNode *> queue;
    TreeNode *root = nullptr;
    vector<vector<int>> ans;

    root = new TreeNode(atoi(argv[1]));
    queue.push_back(root);
    for (int i = 2; i < argc; i += 2) {
        TreeNode *prev = queue.front(), *n;
        int val;

        queue.pop_front();

        if (strcmp(argv[i], "null")) {
            val = atoi(argv[i]);
            n = new TreeNode(val);
            prev->left = n;
            queue.push_back(n);
        }

        if (i == argc - 1)
            break;

        if (strcmp(argv[i+1], "null")) {
            val = atoi(argv[i+1]);
            n = new TreeNode(val);
            prev->right = n;
            queue.push_back(n);
        }
    }

    ans = solution.levelOrderBottom(root);

    for (auto &aa : ans) {
        for (auto a : aa)
            cout << a << " ";
        cout << endl;
    }

    return 0;
}
